tasks.py

```diff
@@ -149,6 +149,7 @@ def type_check(ctx):
         "modal/_utils/shell_utils.py",
         "test/cls_test.py",  # see mypy bug above - but this works with pyright, so we run that instead
         "modal/_container_io_manager.py",
+        "modal/image.py",
```

Comment:
I ran pyright on image.py while developing this to give some more confidence in not breaking in branching of the code that might not be tested, so I fixed the few type errors in that file while I was at it and added it to the allow list here so we check it going forward